### PR TITLE
feat: add global api for to get env,deps in dynamic call

### PR DESCRIPTION
**Pull request by brew0722**
# Description

Closes https://github.com/line/wasmvm/pull/61 https://github.com/line/lbm-sdk/pull/500

Adds a global API instead of the env that was passed as a parameter.
A contract can obtain information by calling the global API.

## Types of changes

- [ ] Bug fix (changes which fixes an issue)
- [x] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/line/cosmwasm/blob/main/CONTRIBUTING.md).
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

**Comment by brew0722** [comment]
I added one more api also for `deps`(https://github.com/line/cosmwasm/pull/182/commits/c15602b92d798edcee90cd7789d89c467f9a3747).
but It has one problem.
It hasn't problem based on product code because deps is stateless.
However, it has problem based on test code because mock deps has state.
But I need deps to implement the sample contract logic of Virtual callstack right now, so I add it first.

File: packages/vm/src/environment.rs [hunk 1]
```diff
@@ -260,6 +260,20 @@ impl<A: BackendApi, S: Storage, Q: Querier> Environment<A, S, Q> {
         });
     }
 
+    pub fn get_serialized_env(&self) -> Vec<u8> {
+        self.with_context_data(|context_data| match &context_data.serialized_env {
+            Some(env) => Ok(env.clone()),
```
Comment (shiki-tak) [inline packages/vm/src/environment.rs:265]: Do we need to clone env? How about to_vec()?

**Review by shiki-tak — APPROVED** [review]
LGTM

File: packages/vm/src/environment.rs (hunk 1 above)
Comment (brew0722) [reply packages/vm/src/environment.rs:265]: yes. we need `env.clone()`. A clone is required to keep ownership of the `serialized_env` in the context.
serialized_env is already initialized after `to_vec()`.

**Review by loloicci — APPROVED** [review]

**Pull request merged**